### PR TITLE
ABLASTR: Pass `GridType` instead of Global

### PR DESCRIPTION
**Pull request by ax3l**
We cannot use global variables from WarpX in the ABLASTR library, otherwise it will not build on its own or in ImpactX.

Follow-up to #4922

### CODE REVIEW

File: Source/ablastr/utils/Enums.H [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    enum struct GridType {
+        Collocated = 0,
+        Staggered = 1,
+        Hybrid = 2
+    };
+
+    /** Mesh-refinement patch
+     *
+     * The current (fine) or lower-level (coarse) resolution.
+     */
```
Comment (ax3l) [inline Source/ablastr/utils/Enums.H]: @EZoni @RemiLehe @oshapoval pls let me know if these doc strings are reasonable or if you would change something :pray:

File: Source/ablastr/utils/Enums.H [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    enum struct GridType {
+        Collocated = 0,
+        Staggered = 1,
+        Hybrid = 2
+    };
+
+    /** Mesh-refinement patch
+     *
+     * The current (fine) or lower-level (coarse) resolution.
```
Comment (RemiLehe) [inline Source/ablastr/utils/Enums.H]:
```suggestion
     * The fine or coarse patch (in terms of spatial resolution) on the same MR level.
     * https://warpx.readthedocs.io/en/latest/theory/amr.html
```

**Review by RemiLehe — APPROVED** [review]

**Review by EZoni — COMMENTED** [review]
Looks good to me, thanks! Can you just confirm that moving `PatchType` to ABLASTR does not require any additional changes (wherever `PatchType` is used in the code) at this time?

**Comment by ax3l** [comment]
Correct, I moved it over because it categorically is general enough for ABLASTR when introducing this new Enums header. I added a `using namespace` in WarpX to avoid any changes at this point caused by it

**Review by EZoni — APPROVED** [review]

**Pull request merged**